x-pack/plugins/reporting/server/routes/common/jobs/get_job_routes.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema, TypeOf } from '@kbn/config-schema';
+import { KibanaRequest, KibanaResponseFactory } from '@kbn/core-http-server';
+import { promisify } from 'util';
+import { getCounters } from '..';
+import { ReportingCore } from '../../..';
+import { ALLOWED_JOB_CONTENT_TYPES } from '../../../../common/constants';
+import { getContentStream } from '../../../lib';
+import { ReportingRequestHandlerContext, ReportingUser } from '../../../types';
+import { handleUnavailable } from '../generate';
+import { jobsQueryFactory } from './jobs_query';
+import { jobManagementPreRouting } from './job_management_pre_routing';
+
+const validate = {
+  params: schema.object({
+    docId: schema.string({ minLength: 3 }),
+  }),
+};
+
+interface HandlerOpts {
+  path: string;
+  user: ReportingUser;
+  context: ReportingRequestHandlerContext;
+  req: KibanaRequest<TypeOf<typeof validate['params']>>;
+  res: KibanaResponseFactory;
+}
+
+export const commonJobsRouteHandlerFactory = (reporting: ReportingCore) => {
```

Comment:
Added these helpers to avoid duplicated code in `x-pack/plugins/reporting/server/routes/internal/management/jobs.ts` and `x-pack/plugins/reporting/server/routes/public/jobs.ts`